andela/travel_tool_front
# 160525336 Filter role dropdown with roles from API

#### What does this PR do?
populates and filters roles dropdown in new Request form as the user types their role.


#### Description of Task to be completed?
When the user is making a new request, To ensure ease of use, it is important for users to be able to search and select their current roles from a pre-populated list that is dynamically updated as the user types in their role.

#### How should this be manually tested?
branch name: ft-user-role-dropdown-160525336
- Clone the repo - ```git clone https://github.com/andela/travel_tool_front.git```
- cd into the branch ```ft-user-role-dropdown-160525336```
- Run the server ```make start``` or ```yarn start```
- Login to get started
- navigate to the new request modal
- as you input the user role the dropdown should change

#### Any background context you want to provide?
Since this feature depends on the API for the contents to be filtered against, it's important to have a mock server mocking the response of the API if it's not yet available

#### What are the relevant pivotal tracker stories?
[# 160525336](https://www.pivotaltracker.com/story/show/160525336)

#### Screenshots (if appropriate)
<img width="793" alt="screen shot 2018-10-04 at 9 30 06 am" src="https://user-images.githubusercontent.com/19623571/46456711-72478600-c7b8-11e8-8edd-5d95baeb8ff6.png">

<img width="615" alt="screen shot 2018-10-04 at 9 30 37 am" src="https://user-images.githubusercontent.com/19623571/46456693-66f45a80-c7b8-11e8-9569-1a5dfa16a580.png">
